Python PULUMI_TEST_MODE gives incorrect errors

If you try to get the current project or stack from Python with `PULUMI_TEST_MODE` set, you get an error message like this:

```
    def get_project() -> str:
        """
        Returns the current project name.
        """
        project = SETTINGS.project
        if not project:
            require_test_mode_enabled()
>           raise RunError('Missing project name; for test mode, please set PULUMI_NODEJS_PROJECT')
E           pulumi.errors.RunError: Missing project name; for test mode, please set PULUMI_NODEJS_PROJECT
venv/lib/python3.7/site-packages/pulumi/runtime/settings.py:132: RunError
```

This is obviously incorrect (it says to use `PULUMI_NODEJS_PROJECT`). However, unlike the Node.js runtime which will respect this environment variable, Python doesn't offer a `PULUMI_PYTHON_PROJECT` equivalent.

The new [runtime mocking](https://github.com/pulumi/pulumi/pull/3738) capability offers a new way of accomplishing this:

```python
pulumi.runtime.set_mocks(None,
    project=os.getenv("PULUMI_PYTHON_PROJECT"),
    stack=os.getenv("PULUMI_PYTHON_STACK"))
```

As part of https://github.com/pulumi/pulumi/issues/2287 we should decide whether to support using an environment variable (it's slightly easier and consistent with the Node.js runtime) but, in any case, we need to ensure the error message is fixed.

--- a/sdk/nodejs/runtime/settings.ts
+++ b/sdk/nodejs/runtime/settings.ts
@@ -139,7 +139,7 @@ export function getProject(): string {
     requireTestModeEnabled();
 
     // And now an error if test mode is enabled, instructing how to manually configure the project:
-    throw new Error("Missing project name; for test mode, please set PULUMI_NODEJS_PROJECT");
+    throw new Error("Missing project name; for test mode, please call `pulumi.runtime.setMocks`");
 }
 
 /** @internal Used only for testing purposes. */

--- a/sdk/python/lib/pulumi/runtime/settings.py
+++ b/sdk/python/lib/pulumi/runtime/settings.py
@@ -129,7 +129,7 @@ def get_project() -> str:
     project = SETTINGS.project
     if not project:
         require_test_mode_enabled()
-        raise RunError('Missing project name; for test mode, please set PULUMI_NODEJS_PROJECT')
+        raise RunError('Missing project name; for test mode, please call `pulumi.runtime.set_mocks`')
     return project
 
 

